nestkernel/connection_manager.cpp

```diff
@@ -1231,13 +1231,19 @@ nest::ConnectionManager::get_connections(
 
           for ( size_t i = 0; i < source_lcids.size(); ++i )
           {
-            conns_in_thread.push_back( ConnectionDatum( ConnectionID(
-              source_table_.get_gid( tid, syn_id, source_lcids[ i ] ),
-              *t_gid,
-              tid,
-              syn_id,
-              source_lcids[ i ] ) ) );
+            if ( synapse_label == UNLABELED_CONNECTION
+              or connections->get_syn_label( source_lcids[ i ] )
+                == synapse_label )
+            {
+              conns_in_thread.push_back( ConnectionDatum( ConnectionID(
```

Comment:
why don't we use `ConnectorBase::get_connection` here? this would also take care of filtering by label and would make this if branch consistent with the others. in addition we wouldn't need to introduce the helper functions below. is speed an issue here?